### PR TITLE
Update flow reconciler

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
QoL improvements to the infrastructure reconciler
```
```other operator
Add migration test for the infrastructure reconciler
```

### CODE REVIEW

File: pkg/webhook/infrastructure/flow.go [hunk 1]
```diff
@@ -44,23 +46,55 @@ func (m *flowMutator) Mutate(ctx context.Context, new, old client.Object) error
...
 	gctx := extensionscontextwebhook.NewGardenContext(m.client, new)
 	cluster, err := gctx.GetCluster(ctx)
 	if err != nil {
 		return err
 	}
 
-	// force terraformer case
-	if metav1.HasAnnotation(cluster.Shoot.ObjectMeta, azure.AnnotationKeyUseTF) {
+	// skip if shoot is being deleted
+	if cluster.Shoot.DeletionTimestamp != nil {
```
[the remark this answers is not in the document]
Comment (kon-angelo) [reply pkg/webhook/infrastructure/flow.go:60]: `old` in this case is the infrastructure object. The check is preventing any mutation to happen if the shoot has been deleted (but the shoot reconciliation has not reached yet the infrastructure).

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Pull request merged**